app/facade/partner_facade.rb

```diff
@@ -0,0 +1,37 @@
+class PartnerFacade
+
+  attr_reader :partner
+
+  def build_partner(partner_params=nil)
+    @partner = Partner.new(permitted_params(partner_params))
+  end
+
+  def load_partner(id)
+    @partner = Partner.find(id)
+  end
+
+  def update_partner(id, partner_params)
+    self.load_partner(id)
+    @partner.attributes = permitted_params(partner_params)
```

Comment:
If you use `@partner.update_attributes= permitted(params)`, would it eliminate the need for the `save` in the controller?